### ISSUE TITLE
Fix ESLint error in weather layers

### ISSUE THREAD

**Issue by VIKTORVAV99** [electricitymaps/electricitymaps-contrib#5121]
Currently ESLint is disabled for most of the weather layer files, we should strive to get ESLint enabled for these and fix the underlying errors instead.

**Comment by qyearsley** [comment]
The files there that have eslint disabled

**Comment by VIKTORVAV99** [comment]
> The files there that have eslint disabled (e.g. https://github.com/electricitymaps/electricitymaps-contrib/blob/master/web/src/features/weather-layers/wind-layer/windy.ts) have comments like:
> 
> ```
> // TODO: Rewrite completely
> // This file was taken from https://github.com/esri/wind-js, and modified
> ```
> 
> So, does this mean we want to refactor and keep the logic and behavior the same, but improve readability and style? Asking because I didn't want to potentially waste time trying to improve what's there if someone's gonna rewrite completely.

Hi @qyearsley!

Yeah we want to rewrite these but they seems to be really fragile so nobody has really taken the steps to do it yet. And yes the inputs and outputs need to stay the same.

If you want to take a crack at it I think the rest of the team will be very welcome to it as well but keep in mind this might me a lot more difficult than it initially seems.

So if you wish I'll happily assign you the issue if you want to work on it.

CC @madsnedergaard and @tonypls for visability.

**Comment by qyearsley** [comment]
Makes sense, thanks.

It sounds like verifying that nothing is broken after each incremental change (even if it appears like it shouldn't affect behavior) would be a good idea; there are not currently unit tests for wind-layer/windy.ts or wind-layer/scales.ts; what's the most useful thing I can do if I try to make a change and want to verify that it should still work? `pnpm run mockserver` or something else?

**Comment by VIKTORVAV99** [comment]
> Makes sense, thanks.
> 
> It sounds like verifying that nothing is broken after each incremental change (even if it appears like it shouldn't affect behavior) would be a good idea; there are not currently unit tests for wind-layer/windy.ts or wind-layer/scales.ts; what's the most useful thing I can do if I try to make a change and want to verify that it should still work? `pnpm run mockserver` or something else?

Honestly I'm not sure, we don't have any mock data for the solar or wind layers and contributors don't have access to the API for local development. 

I think it maybe it would be best to create unit tests first and then refractor the code. These tests should just check input and output as the internal code would change. I can probably add a snapshot of the weather data if you wish to proceed with this.

**Comment by qyearsley** [comment]
I agree; it would be helpful to have something like a snapshot to serve as mock data for a test here.

**Comment by VIKTORVAV99** [comment]
So just to provide an update on this:

I have attempted to add mockdata for the wind and solar layers and while I got the mockserver to serve 1 of the two needed files I can't get it to serve both (hour 1 and hour 2) due the the fact it uses query params. 

So I think we need to rework things internally so this only returns 1 file (with both hours of data) before we proceed with rewriting the weather layers, there are obvious advantages to reducing the http calls alone that makes it worth it long term.

But before we do so we need to shape the new data structure and determine what exactly we need to return in every request.
